### PR TITLE
New visualization contribution guidelines

### PR DESCRIPTION
**Pull request by kzadurska**

### CODE REVIEW

**Comment by kzadurska** [comment]
Whoops, sorry for the force push, wanted to have one commit but forgot to save last changes 🙈

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,5 +1,146 @@
...
+
+#### Set up a new type
+Add your new `Visualization` type in `src/common/models/visualization-manifest/visualization-manifest.ts`.
+
+#### Add a manifest
+Add a new entry in `src/common/visualization-manifests/index.ts`.
+
+Add a new instance of `VisualizationManifest` in `src/common/visualization-manifests/<visualizationName>/<visualizationName>.ts`:
+you can use `emptySettingsConfig` for `visualizationSettings` in the beginning and add settings later if needed.
+Write an `evaluateRules` function to make sure your visualization is shown under certain conditions.
```
Comment (kzadurska) [inline CONTRIBUTING.md:23]: Add example function, this is not the easiest part of the process.

Comment (kzadurska) [reply CONTRIBUTING.md]: Added an example 3ac9630

**Review by mkuthan — APPROVED** [review]
Very well written, thanks a lot!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,5 +1,156 @@
...
 - Bug reports, feature requests, and any sort of feedback is very welcome.
 - Pull requests fixing bugs are very welcome as well.
-- For pull requests involving any changes to the UI, please discuss them first in a github issue.
+- For pull requests involving any changes to the UI, please discuss them first in a GitHub issue.
+
+## Add new visualization to Turnilo
+
+### Initial setup
+
+_[Example code for scatterplot setup](https://github.com/allegro/turnilo/pull/827)_
```
[the remark this answers is not in the document]
Comment (adrianmroz-allegro) [reply CONTRIBUTING.md:13]: > I'm wondering about closed PR retention on GitHub. Do they stay forever?
With squash merges it shouldn't be just one commit? Commits should live forever, so maybe link to it?

[the remark this answers is not in the document]
Comment (kzadurska) [reply CONTRIBUTING.md:13]: > I'm wondering about closed PR retention on GitHub. Do they stay forever?

As far as I know pull requests are for ever and do not disappear. If someone wants to, they can even restore a deleted pull request branch.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,5 +1,156 @@
...
+
+#### Set up a new type
+Add your new `Visualization` type in `src/common/models/visualization-manifest/visualization-manifest.ts`.
```
[the remark this answers is not in the document]
Comment (kzadurska) [reply CONTRIBUTING.md:16]: - [ ] Sure, I'll add links to files.

[the remark this answers is not in the document]
Comment (kzadurska) [reply CONTRIBUTING.md]: 7f3d4d6df76a done!

File: CONTRIBUTING.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (kzadurska) [reply CONTRIBUTING.md:13]: > I would start with the explicit code fragments (sections below) and show the complete PR at the end of the "initial setup" section.

@mkuthan Should I add more code examples? Or just move scatterplot links to ends of sections?

[the remark this answers is not in the document]
Comment (kzadurska) [reply CONTRIBUTING.md:13]: I assume it is okay as is 😉

**Pull request merged**